File: String-2/catDog.java
public boolean catDog(String str) {
  int cat = 0, dog = 0;
  
  if(str.length() <= 3) {
    if(str.length() < 3) return true;
    else {
      return (str.equals("cat") || str.equals("dog"))? false : true;
    }
  }
  for(int i = 0; i < str.length() - 2; i++) {
    if(str.substring(i, i + 3).equals("cat")) cat++;
    if(str.substring(i, i + 3).equals("dog")) dog++;
  }
  return cat == dog;
}

File: python/String-2/cat_dog.py
def cat_dog(str):
  cat_count = 0
  dog_count = cat_count

  for i in range(len(str)-2):
      if str[i:i+3] == "cat":
          cat_count += 1
      if str[i:i+3] == "dog":
          dog_count += 1

  return cat_count == dog_count
